src/user/client.ts

```diff
@@ -35,6 +33,21 @@ function createUserObject(user: firebase.User): User {
   return newUser;
 }
 
+function loadGoogleAuth(): Promise<gapi.auth2.GoogleAuth> {
```

Comment:
I extracted the common parts from `googleLogin()` and `getAccessToken()` to here, which seems a bit similar to `Auth.currentUser()` in firebase now. I still call it in these two functions, so it's not one per load (for example, one per user login) because we are not saving gapi-related info locally between these two functions (what I mean by this is, for example, after calling `googleLogin()`, we are not storing the access token we get as a local variable to access later. The reason I did not do this is (I tried this before), the access token expires after 1 hour, so the only way I could find to get a valid / up-to-date access token was through calling `gapi.auth2.getAuthInstance().currentUser`, the logic in `getAccessToken()`. Does this look reasonable? Or, I think maybe we could look into storing the current user in state, for example https://vuejs.org/v2/guide/state-management.html, that could for example update whenever `gapi.auth2.getAuthInstance().currentUser` changes, which might prevent us from calling `loadGoogleAuth` manually multiple times, though I am not 100% sure how this would work without trying. I can also keep this in mind as a potential future improvement if there's time for it later on.

Reply:
Yup this is ok, thanks!